checker/src/org/checkerframework/checker/index/upperbound/UpperBoundChecker.java

```diff
@@ -17,6 +18,23 @@
 @SuppressWarningsKeys({"index", "upperbound"})
 public class UpperBoundChecker extends BaseTypeChecker {
 
+    private HashSet<String> collectionBaseClasses;
+
+    public UpperBoundChecker() {
+        collectionBaseClasses = new HashSet<>(3);
+        collectionBaseClasses.add("java.util.List");
```

Comment:
How about using the `.class` literals to get the correct name? This code is executed only once, so it shouldn't have a performance impact.